src/plugins/core.js

```diff
@@ -170,74 +163,64 @@ function Plugin(options = {}) {
       (chars.equals(nextChars))
     )
 
-    // Add the `isNative` flag directly, so we don't have to re-transform.
+    // If `isNative`, set the flag on the transform.
     if (isNative) {
-      next = next.set('isNative', isNative)
+      transform.setIsNative(true)
     }
 
-    // If not native, prevent default so that the DOM remains untouched.
-    if (!isNative) e.preventDefault()
+    // Otherwise, prevent default so that the DOM remains untouched.
+    else {
+      e.preventDefault()
+    }
 
     debug('onBeforeInput', { data, isNative })
-    return next
   }
 
   /**
    * On blur.
    *
    * @param {Event} e
    * @param {Object} data
-   * @param {State} state
-   * @return {State}
+   * @param {Transform} transform
    */
 
-  function onBlur(e, data, state) {
+  function onBlur(e, data, transform) {
     debug('onBlur', { data })
-    return state
-      .transform()
-      .blur()
-      .apply()
+    transform.blur()
   }
 
   /**
    * On copy.
    *
    * @param {Event} e
    * @param {Object} data
-   * @param {State} state
-   * @return {State}
+   * @param {Transform} transform
    */
 
-  function onCopy(e, data, state) {
+  function onCopy(e, data, transform) {
     debug('onCopy', data)
-    onCutOrCopy(e, data, state)
+    onCutOrCopy(e, data, transform)
   }
 
   /**
    * On cut.
    *
    * @param {Event} e
    * @param {Object} data
-   * @param {State} state
+   * @param {Transform} transform
    * @param {Editor} editor
-   * @return {State}
    */
 
-  function onCut(e, data, state, editor) {
+  function onCut(e, data, transform, editor) {
     debug('onCut', data)
-    onCutOrCopy(e, data, state)
+    onCutOrCopy(e, data, transform)
     const window = getWindow(e.target)
 
     // Once the fake cut content has successfully been added to the clipboard,
     // delete the content in the current selection.
     window.requestAnimationFrame(() => {
-      const next = editor
-        .getState()
-        .transform()
-        .delete()
-        .apply()
-
-      editor.onChange(next)
+      const t = editor.transform().delete()
```

Comment:
editor.transform need to have a function callback and editor.applyTransform has been removed on the newest changes